packages/braid-design-system/src/lib/components/Button/Button.tsx

```diff
@@ -355,21 +357,42 @@ export const ButtonText = ({
         size={size}
         baseline={false}
       >
-        {icon ? (
+        {icon && iconPosition === 'leading' ? (
           <Box
             component="span"
             paddingRight="xsmall"
-            className={
+            className={[
+              styles.nowrap,
               shouldReducePaddingX || bleed
                 ? null
-                : negativeMargin('left', 'xxsmall')
-            }
+                : negativeMargin('left', 'xxsmall'),
+            ]}
           >
             {icon}
+            {
+              '⁠' /* Word joiner character, a zero-width non-breaking character to prevent the icon wrapping onto its own line */
+            }
           </Box>
         ) : null}
         {children}
         {loading ? <ButtonLoader /> : null}
+        {!loading && icon && iconPosition === 'trailing' ? (
```

Comment:
What's the reasoning behind not showing a trailing icon when `loading`?

Reply:
Presented a strange UX when combining both, i.e. the animated ellipsis followed by an icon. Ideally I'd like to use the trailing icon slot for a better loading indicator. What are your thoughts?

Reply:
Yeah I was thinking the behaviour looked a bit weird, but don't really have a better solution. I think replacing the icon with a loading icon is a good idea.